tests/Domain/Integrations/Repositories/EloquentIntegrationRepositoryTest.php

```diff
@@ -561,4 +561,28 @@ public function test_it_can_save_partner_status(): void
             'partner_status' => IntegrationPartnerStatus::FIRST_PARTY,
         ]);
     }
+
+    // https://jira.publiq.be/browse/PPF-555
+    public function test_it_can_save_integration_uitpas_always_with_key_visibility_v2(): void
+    {
+        $integrationId = Uuid::uuid4();
+
+        $searchIntegration = new Integration(
+            $integrationId,
+            IntegrationType::UiTPAS,
+            'First party integration',
+            'First party integration',
+            Uuid::uuid4(),
+            IntegrationStatus::Draft,
+            IntegrationPartnerStatus::FIRST_PARTY,
+        );
+        $searchIntegration = $searchIntegration->withKeyVisibility(KeyVisibility::v1);
```

Comment:
I'm wondering if this should maybe throw an exception instead of magically setting the visibility differently than expected